### PR TITLE
[incubator/Elasticsearch] Allow for enabling monitoring for 6.3+

### LINKED ISSUE #7901: [incubator/elasticsearch] chart does not support new xpack syntax
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Which chart**: [incubator/elasticsearch]

**What happened**: in elasticsearch 6.3+ xpack is installed by default and the config parameters to enable changed. 

**What you expected to happen**:  We need elasticsearch to allow the enabling of xpack systems

**How to reproduce it** (as minimally and precisely as possible): set `cluster.xpackEnable true`

**Anything else we need to know**: I am testing out fixes and will submit a PR once my testing is completed.

### PR DESCRIPTION
**Pull request by aba182**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
    - fixes #7901

#### Special notes for your reviewer:
To test the new functionality please deploy with the changes in the README otherwise the default options are specified. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

File: incubator/elasticsearch/README.md [hunk 1]
```diff
@@ -189,11 +202,3 @@ Create cluster with Storage class `ssd` on Kubernetes 1.5+
 ```
 $ helm install incubator/elasticsearch --name my-release --set data.storageClass=ssd,data.storage=100Gi
 ```
-
```
Comment (rendhalver) [inline incubator/elasticsearch/README.md]: I am guessing this removal was unintentional?

Comment (aba182) [reply incubator/elasticsearch/README.md]: That is correct, let me removed for cleanliness.

**Comment by catmeme** [comment]
Thanks so much, we ran into this same exact issue today!

**Pull request merged**